pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnectionMetaData.java

```diff
@@ -150,30 +150,34 @@ public ResultSet getTableTypes()
   public ResultSet getColumns(String catalog, String schemaPattern, String tableNamePattern, String columnNamePattern)
       throws SQLException {
 
+    if (tableNamePattern != null && tableNamePattern.equals("%")) {
```

Comment:
Is there a scenario where the `%` value is sent? Do we need to catch more incompatible patterns here?

Reply:
the javadoc for `java.sql.DatabaseMetaData` specifies how/why...

```
Some DatabaseMetaData methods take arguments that are String patterns. These arguments all have names such as fooPattern. Within a pattern String, "%" means match any substring of 0 or more characters, and "_" means match any one character. Only metadata entries matching the search pattern are returned. If a search pattern argument is set to null, that argument's criterion will be dropped from the search.
```

As it relates to IntelliJ/Jetbrains there is a "Diagnostic Introspector" which shows the results for all the metadata methods called and one of those calls passes `%` for table name.  No other patterns are passed other than exact table name.